### PR TITLE
Refactoring API round 1

### PR DESCRIPTION
**Pull request by alexjoybc**
# Description

This PR includes the following proposed change(s):

- create submission objects
- refactor get account
- removed unused pii

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation

if your change is a breaking change, please add `breaking change` label to this PR

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by akroon3r** [comment]
ROUND 1!?!

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        }
+
+        @Override
+        public List<DocumentType> getDocumentTypes(String courtLevel, String courtClass) {
+            return null;
+        }
+    }
+
+
+}
```
Comment (TayGov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/SubmissionConfigTest.java:152]: Newline

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/TestHelpers.java [hunk 2]
```diff
@@ -42,22 +46,13 @@
     public static final String PARTY_TYPE_CD = "PARTYCD";
     public static final String ROLE_TYPE_CD = "ROLECD";
 
-    public static InitialPackage createInitalPackage(Court court, List<DocumentProperties> documentProperties) {
+    public static InitialPackage createInitalPackage(ca.bc.gov.open.jag.efilingapi.api.model.Court court, List<DocumentProperties> documentProperties) {
```
Comment (TayGov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/TestHelpers.java:49]: I wonder if we need to start prefixing the models. So don't collisions like this. Although I'm not a fan of it either....

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/models/FilingPakageTest.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                    LOCATION_DESCRIPTION,
+                    LEVEL,
+                    LEVEL_DESCRIPTION,
+                    COURT_CLASS,
+                    CLASS_DESCRIPTION,
+                    DIVISION,
+                    FILE_NUMBER,
+                    PARTICIPATING_CLASS);
+    }
+}
```
Comment (TayGov) [inline src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/submission/models/FilingPakageTest.java:131]: Newline

File: src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/config/AutoConfigurationTest.java [hunk 4]
```diff
@@ -23,35 +23,35 @@
...
+                        "jag.efiling.soap.clients[4].password=password",
+                        "jag.efiling.soap.clients[4].uri=http://locahost",
+                        "jag.efiling.soap.clients[5].client=FILING",
+                        "jag.efiling.soap.clients[5].userName=username",
+                        "jag.efiling.soap.clients[5].password=password",
+                        "jag.efiling.soap.clients[5].uri=http://locahost",
+                        "jag.efiling.soap.clients[6].client=SERVICE",
+                        "jag.efiling.soap.clients[6].userName=username",
+                        "jag.efiling.soap.clients[6].password=password",
+                        "jag.efiling.soap.clients[6].uri=http://locahost")
```
Comment (TayGov) [inline src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/config/AutoConfigurationTest.java:54]: Nice

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/account/service/AccountServiceImpl.java [hunk 5]
```diff
@@ -1,19 +1,25 @@
...
+    private final CreateAccountRequestMapper createAccountRequestMapper;
 
-    public AccountServiceImpl(EfilingAccountService efilingAccountService) {
+    public AccountServiceImpl(EfilingAccountService efilingAccountService, CreateAccountRequestMapper createAccountRequestMapper) {
         this.efilingAccountService = efilingAccountService;
+        this.createAccountRequestMapper = createAccountRequestMapper;
     }
 
     @Override
+    @Cacheable(cacheNames = "accountDetails", key = "#universalId", cacheManager = "accountDetailsCacheManager", unless = "#result == null || #result.getInternalClientNumber() == null")
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/account/service/AccountServiceImpl.java:22]: #universalID ?

File: src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/models/Document.java [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+        public Builder mimeType(String mimeType) {
+            this.mimeType = mimeType;
+            return this;
+        }
+
+        public Document create() {
+            return new Document(this);
+        }
+
+
```
Comment (akroon3r) [inline src/backend/efiling-api/src/main/java/ca/bc/gov/open/jag/efilingapi/submission/models/Document.java:155]:
```suggestion
```

File: src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/config/AutoConfigurationTest.java (hunk 4 above)
Comment (akroon3r) [reply src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/config/AutoConfigurationTest.java:54]: 🦈

File: src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/TestHelpers.java (hunk 2 above)
Comment (alexjoybc) [reply src/backend/efiling-api/src/test/java/ca/bc/gov/open/jag/efilingapi/TestHelpers.java:49]: as we spoke, we keep it the same for now ;-)

File: src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/config/AutoConfigurationTest.java (hunk 4 above)
Comment (alexjoybc) [reply src/backend/libs/efiling-cso-starter/src/test/java/ca/bc/gov/open/jag/efilingcsostarter/config/AutoConfigurationTest.java:54]: lol

**Pull request merged**